CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Simulation/CoronavirusSteppables.py

```diff
@@ -450,42 +465,48 @@ def step(self, mcs):
+                        for y in range(yi, yi + int(cell_diameter)):
+                            cell = self.cellField[x, y, 1]
+                            if cell:
+                                open_space = False
+                                break
+                    if open_space:
+                        concentration_iteration = self.field.Virus[xi, yi, 1]
+                        if concentration_iteration >= viral_concentration:
+                            viral_concentration = concentration_iteration
+                            x_seed = xi
+                            y_seed = yi
                 if open_space:
-                    concentration_iteration = self.field.Virus[xi, yi, 1]
-                    if concentration_iteration >= viral_concentration:
-                        viral_concentration = concentration_iteration
-                        x_seed = xi
-                        y_seed = yi
-            if open_space:
-                cell = self.new_cell(self.IMMUNECELL)
-                cell.dict['activated'] = False  # flag for immune cell being naive or activated
-                # cyttokine params
-                cell.dict['ck_production'] = max_ck_secrete_im  # TODO: replace secretion by hill
-                cell.dict['ck_consumption'] = max_ck_consume  # TODO: replace by hill
-                cell.dict['tot_ck_upt'] = 0
-
-                self.cellField[x_seed:x_seed + int(cell_diameter), y_seed:y_seed + int(cell_diameter), 1] = cell
-                cd = self.chemotaxisPlugin.addChemotaxisData(cell, "cytokine")
-                if cell.dict['activated']:
-                    cd.setLambda(50.0)
-                else:
-                    cd.setLambda(0.0)
-                cd.assignChemotactTowardsVectorTypes([self.MEDIUM])
-                cell.targetVolume = cell_volume
-                cell.lambdaVolume = cell_volume
+                    cell = self.new_cell(self.IMMUNECELL)
+                    cell.dict['activated'] = False  # flag for immune cell being naive or activated
+                    # cytokine parameters
+                    cell.dict['ck_production'] = max_ck_secrete_im  # TODO: replace secretion by hill
+                    cell.dict['ck_consumption'] = max_ck_consume  # TODO: replace by hill
+                    cell.dict['tot_ck_upt'] = 0
+
+                    self.cellField[x_seed:x_seed + int(cell_diameter), y_seed:y_seed + int(cell_diameter), 1] = cell
+                    cd = self.chemotaxisPlugin.addChemotaxisData(cell, "cytokine")
+                    if cell.dict['activated']:
+                        cd.setLambda(lamda_chemotaxis)
+                    else:
+                        cd.setLambda(0.0)
+                    cd.setLambda(lamda_chemotaxis)
+                    cd.assignChemotactTowardsVectorTypes([self.MEDIUM])
+                    cell.targetVolume = cell_volume
+                    cell.lambdaVolume = cell_volume
```

Comment:
lambda_volume = target_volume?? This was like this on the previous version, so we should really check for it. Do we have lambda_volume as a parameter somewhere?

Reply:
So I defined lambda volume to be same at target volume based on Macieks paper on cancer, but that number can be changed as its arbitrary.

Reply:
Interesting. ok

Reply:
But I think we can make it a more user adaptable by defining it on the heading (with all the other CompuCell parameters)